common/main/java/com/couchbase/lite/Collection.java

```diff
@@ -0,0 +1,189 @@
+package com.couchbase.lite;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import com.couchbase.lite.internal.utils.Preconditions;
+
+
+public class Collection implements Indexable, DatabaseChangeObservable {
+    private final String name;
+    private final int count;
+    private final Scope scope;
+    private final Database database;
+
+    public Collection(@NonNull String name, int count, @NonNull Scope scope, @NonNull Database database) {
```

Comment:
No... count doesn't belong in here...  Just delegate it like you do the rest of the properties.
... also, right now, the name is *always* "_default", and the scope is always the default scope (not sure what it's name is).  This constructor should take only one argument: the database